parts/xmem_nozzle.go

```diff
@@ -1554,15 +1540,15 @@ func (xmem *XmemNozzle) batchGetHandler(count int, finch chan bool, return_ch ch
 				}
 
 				if !isNetTimeoutError(err) && err != PartStoppedError {
-					logger.Errorf("%v %v received fatal error and had to abort. Expected %v responses, got %v responses. err=%v", xmem.Id(), batchGetStr, count, len(respMap), err)
+					logger.Errorf("%v batchGet received fatal error and had to abort. Expected %v responses, got %v responses. err=%v", xmem.Id(), count, len(respMap), err)
 					logger.Infof("%v Expected=%v, Received=%v\n", xmem.Id(), opaque_keySeqno_map.CloneAndRedact(), base.UdTagBegin, respMap, base.UdTagEnd)
 				} else {
-					logger.Errorf("%v %v timed out. Expected %v responses, got %v responses", xmem.Id(), batchGetStr, count, len(respMap))
+					logger.Errorf("%v batchGet timed out. Expected %v responses, got %v responses", xmem.Id(), count, len(respMap))
 					logger.Infof("%v Expected=%v, Received=%v\n", xmem.Id(), opaque_keySeqno_map.CloneAndRedact(), base.UdTagBegin, respMap, base.UdTagEnd)
 				}
 				return
-
 			} else {
+				isGetMeta := response.Opcode == base.GET_WITH_META
```

Comment:
Make a function `decodeOpcodeForGet(response)` similar in functionality to `composeRequestForGet`
In that function, write a select, return `GetMetaReceived` or `GetDocReceived`, or panic if default

Here, just have a simple line:
`receivedEvent := decodeOpCodeForGet(response)`

Can remove `if` and `else` from line 1571 to 1575 to make code cleaner

Reply:
Line 1610 and line 1582 were added as part of the CCR code before the refactoring effort
https://review.couchbase.org/c/goxdcr/+/135094

Do we need `!isGetMeta`? Can they be removed?

Reply:
I am not sure how it would impact the normal getMeta. I think functionally it would be OK to remove. Not sure about performance. These are only needed for subdoc_lookup.